### PR TITLE
Fix linking of vendored libraries in pod targets

### PR DESCRIPTION
**Pull request by Westacular**
Closes [8453](https://github.com/CocoaPods/CocoaPods/issues/8453).

Also, updates unit tests to avoid this in future.

Specifically: the build_settings unit tests _should_ have caught this issue, however, the banana-lib test project that tested the relevant functionality had both Bananalib.framework and libBananalib.a, which both have the same name when stripped down, creating an ambiguity.

I've updated the banana-lib project to instead have BananaFramework.framework and libBananaStaticLib.a, and updated all the unit tests in accordance with the new names.

I've verified that the relevant build_setting test now fails without the fix, and passes with it.

One curiosity was that the `'detects duplicate library names'` test was failing following the rename, and not raising a warning as expected. Not entirely sure why that is... in any case, I've modified that test (based on the example set in `'detects duplicate framework names'`) and it now passes again.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,8 +12,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
-* None.  
-
+* Fix linking of vendored libraries and frameworks in pod targets
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit please add 2 empty spaces in the end of this line for formatting.

Comment (Westacular) [reply CHANGELOG.md]: Oh, right. Done.

File: lib/cocoapods/target/build_settings.rb [hunk 2]
```diff
@@ -607,7 +607,7 @@ def initialize(target, test_spec = nil)
 
           libraries = []
           if test_xcconfig? || target.requires_frameworks? && !target.static_framework?
-            libraries.concat vendored_static_frameworks.map { |l| File.basename(l, '.framework') }
+            libraries.concat vendored_static_libraries.map { |l| File.basename(l, l.extname).sub(/\Alib/, '') }
```
Comment (amorde) [inline lib/cocoapods/target/build_settings.rb]: the logic of `MyLibrary` -> `libMyLibrary.a` also existing in `:static_libraries_to_import` and `:dynamic_libraries_to_import` - could you pull it out into a function?

File: spec/unit/target/build_settings/pod_target_settings_spec.rb [hunk 3]
```diff
@@ -221,7 +221,7 @@ class BuildSettings
             @generator.spec_consumers.each { |sc| sc.stubs(:frameworks => []) }
             @generator.stubs(:dependent_targets => [pod_target])
             @generator.other_ldflags.should.
-              be == %w(-l"Bananalib" -l"xml2" -framework "Bananalib")
+              be == %w(-l"BananaStaticLib" -l"xml2" -framework "BananaFramework")
```
Comment (amorde) [inline spec/unit/target/build_settings/pod_target_settings_spec.rb]: nice catch on this test!

File: lib/cocoapods/target/build_settings.rb (hunk 2 above)
Comment (Westacular) [reply lib/cocoapods/target/build_settings.rb]: Sure, done

**Comment by dnkoutso** [comment]
thanks!

**Pull request merged**